internal/controllers/v1beta1/garbagecollection/garbagecollection.go

```diff
@@ -108,9 +113,12 @@ func (c *GarbageCollectionController) Reconcile(ctx context.Context, request rec
 			return reconcile.Result{
 				Requeue: true,
 			}, err
+		} else {
+			delete(c.managedCerts, request.String())
```

Comment:
nit: The `else` isn't necessary since if it errors on delete, the return exits this block